In [14]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
# Reading in the data
df=pd.read_csv("co2_emissions_from_agri.csv")

# Data information

In [3]:

print ("The data types in this dataset are:\n",df.dtypes,"\n")
print("The dataset has",len(df.select_dtypes(exclude=['float64', 'int64']).columns),"categorical variables. \nThe dataset has",len(df.select_dtypes(include=['float64', 'int64']).columns),"numerical_variables \n")

print ("The shape of the dataset is:\n",df.shape)

The data types in this dataset are:
 Area                                object
Year                                 int64
Savanna fires                      float64
Forest fires                       float64
Crop Residues                      float64
Rice Cultivation                   float64
Drained organic soils (CO2)        float64
Pesticides Manufacturing           float64
Food Transport                     float64
Forestland                         float64
Net Forest conversion              float64
Food Household Consumption         float64
Food Retail                        float64
On-farm Electricity Use            float64
Food Packaging                     float64
Agrifood Systems Waste Disposal    float64
Food Processing                    float64
Fertilizers Manufacturing          float64
IPPU                               float64
Manure applied to Soils            float64
Manure left on Pasture             float64
Manure Management                  float64
Fires in organic 

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6965 entries, 0 to 6964
Data columns (total 31 columns):
 #   Column                           Non-Null Count  Dtype  
---  ------                           --------------  -----  
 0   Area                             6965 non-null   object 
 1   Year                             6965 non-null   int64  
 2   Savanna fires                    6934 non-null   float64
 3   Forest fires                     6872 non-null   float64
 4   Crop Residues                    5576 non-null   float64
 5   Rice Cultivation                 6965 non-null   float64
 6   Drained organic soils (CO2)      6965 non-null   float64
 7   Pesticides Manufacturing         6965 non-null   float64
 8   Food Transport                   6965 non-null   float64
 9   Forestland                       6472 non-null   float64
 10  Net Forest conversion            6472 non-null   float64
 11  Food Household Consumption       6492 non-null   float64
 12  Food Retail         

In [5]:
df.head()

Area  Year  Savanna fires  Forest fires  Crop Residues  \
0  Afghanistan  1990        14.7237        0.0557       205.6077   
1  Afghanistan  1991        14.7237        0.0557       209.4971   
2  Afghanistan  1992        14.7237        0.0557       196.5341   
3  Afghanistan  1993        14.7237        0.0557       230.8175   
4  Afghanistan  1994        14.7237        0.0557       242.0494   

   Rice Cultivation  Drained organic soils (CO2)  Pesticides Manufacturing  \
0            686.00                          0.0                 11.807483   
1            678.16                          0.0                 11.712073   
2            686.00                          0.0                 11.712073   
3            686.00                          0.0                 11.712073   
4            705.60                          0.0                 11.712073   

   Food Transport  Forestland  ...  Manure Management  Fires in organic soils  \
0         63.1152   -2388.803  ...           319.1763                     0.0   
1         61.2125   -2388.803  ...           342.3079                     0.0   
2         53.3170   -2388.803  ...           349.1224                     0.0   
3         54.3617   -2388.803  ...           352.2947                     0.0   
4         53.9874   -2388.803  ...           367.6784                     0.0   

   Fires in humid tropical forests  On-farm energy use  Rural population  \
0                              0.0                 NaN         9655167.0   
1                              0.0                 NaN        10230490.0   
2                              0.0                 NaN        10995568.0   
3                              0.0                 NaN        11858090.0   
4                              0.0                 NaN        12690115.0   

   Urban population  Total Population - Male  Total Population - Female  \
0         2593947.0                5348387.0                  5346409.0   
1         2763167.0                5372959.0                  5372208.0   
2         2985663.0                6028494.0                  6028939.0   
3         3237009.0                7003641.0                  7000119.0   
4         3482604.0                7733458.0                  7722096.0   

   total_emission  Average Temperature °C  
0     2198.963539                0.536167  
1     2323.876629                0.020667  
2     2356.304229               -0.259583  
3     2368.470529                0.101917  
4     2500.768729                0.372250  

[5 rows x 31 columns]

In [6]:
df.describe()

Year  Savanna fires  Forest fires  Crop Residues  \
count  6965.000000    6934.000000   6872.000000    5576.000000   
mean   2005.124910    1188.390893    919.302167     998.706309   
std       8.894665    5246.287783   3720.078752    3700.345330   
min    1990.000000       0.000000      0.000000       0.000200   
25%    1997.000000       0.000000      0.000000      11.006525   
50%    2005.000000       1.651850      0.517900     103.698200   
75%    2013.000000     111.081400     64.950775     377.640975   
max    2020.000000  114616.401100  52227.630600   33490.074100   

       Rice Cultivation  Drained organic soils (CO2)  \
count       6965.000000                  6965.000000   
mean        4259.666673                  3503.228636   
std        17613.825187                 15861.445678   
min            0.000000                     0.000000   
25%          181.260800                     0.000000   
50%          534.817400                     0.000000   
75%         1536.640000                   690.408800   
max       164915.255600                241025.069600   

       Pesticides Manufacturing  Food Transport     Forestland  \
count               6965.000000     6965.000000    6472.000000   
mean                 333.418393     1939.581760  -17828.285678   
std                 1429.159367     5616.748808   81832.210543   
min                    0.000000        0.000100 -797183.079000   
25%                    6.000000       27.958600   -2848.350000   
50%                   13.000000      204.962800     -62.920000   
75%                  116.325487     1207.000900       0.000000   
max                16459.000000    67945.765000  171121.076000   

       Net Forest conversion  ...  Manure Management  Fires in organic soils  \
count           6.472000e+03  ...        6037.000000             6965.000000   
mean            1.760564e+04  ...        2263.344946             1210.315532   
std             1.011575e+05  ...        7980.542461            22669.847760   
min             0.000000e+00  ...           0.432900                0.000000   
25%             0.000000e+00  ...          37.632100                0.000000   
50%             4.444000e+01  ...         269.856300                0.000000   
75%             4.701746e+03  ...        1126.818900                0.000000   
max             1.605106e+06  ...       70592.646500           991717.543100   

       Fires in humid tropical forests  On-farm energy use  Rural population  \
count                      6810.000000         6009.000000      6.965000e+03   
mean                        668.452931         3008.982252      1.785774e+07   
std                        3264.879486        12637.864430      8.901521e+07   
min                           0.000000            0.031900      0.000000e+00   
25%                           0.000000           13.291900      9.731100e+04   
50%                           0.000000          141.096300      1.595322e+06   
75%                           9.577875         1136.925400      8.177340e+06   
max                       51771.256800       248879.176900      9.000991e+08   

       Urban population  Total Population - Male  Total Population - Female  \
count      6.965000e+03             6.965000e+03               6.965000e+03   
mean       1.693230e+07             1.761963e+07               1.732447e+07   
std        6.574362e+07             7.603993e+07               7.251711e+07   
min        0.000000e+00             2.500000e+02               2.700000e+02   
25%        2.173860e+05             2.013260e+05               2.078900e+05   
50%        2.357581e+06             2.469660e+06               2.444135e+06   
75%        8.277123e+06             9.075924e+06               9.112588e+06   
max        9.020778e+08             7.435866e+08               7.133419e+08   

       total_emission  Average Temperature °C  
count    6.965000e+03             6965.000000  
mean     6.409124e+04                0.872989  
std      2.283130e+05

### Missing values

In [7]:
df.isnull().sum()

Area                                  0
Year                                  0
Savanna fires                        31
Forest fires                         93
Crop Residues                      1389
Rice Cultivation                      0
Drained organic soils (CO2)           0
Pesticides Manufacturing              0
Food Transport                        0
Forestland                          493
Net Forest conversion               493
Food Household Consumption          473
Food Retail                           0
On-farm Electricity Use               0
Food Packaging                        0
Agrifood Systems Waste Disposal       0
Food Processing                       0
Fertilizers Manufacturing             0
IPPU                                743
Manure applied to Soils             928
Manure left on Pasture                0
Manure Management                   928
Fires in organic soils                0
Fires in humid tropical forests     155
On-farm energy use                  956


In [8]:
## Dropping rows where missing value is more than 10% of the data set total rows
# Identify columns with more than 10% missing values
cols_to_drop = df.columns[df.isnull().sum() > len(df) * 0.1]

# Drop those columns
df_filtered = df.drop(columns=cols_to_drop)


In [9]:
df_filtered.isnull().sum()

Area                                 0
Year                                 0
Savanna fires                       31
Forest fires                        93
Rice Cultivation                     0
Drained organic soils (CO2)          0
Pesticides Manufacturing             0
Food Transport                       0
Forestland                         493
Net Forest conversion              493
Food Household Consumption         473
Food Retail                          0
On-farm Electricity Use              0
Food Packaging                       0
Agrifood Systems Waste Disposal      0
Food Processing                      0
Fertilizers Manufacturing            0
Manure left on Pasture               0
Fires in organic soils               0
Fires in humid tropical forests    155
Rural population                     0
Urban population                     0
Total Population - Male              0
Total Population - Female            0
total_emission                       0
Average Temperature °C   

### Imputing null values

In [11]:
# imputing all columns with the mean value
numerical_columns = df_filtered.select_dtypes(include=['float64', 'int64']).columns

for col in numerical_columns:
    df_filtered[col] = df_filtered[col].fillna(df_filtered[col].mean())



In [12]:
df_filtered.isnull().sum()

Area                               0
Year                               0
Savanna fires                      0
Forest fires                       0
Rice Cultivation                   0
Drained organic soils (CO2)        0
Pesticides Manufacturing           0
Food Transport                     0
Forestland                         0
Net Forest conversion              0
Food Household Consumption         0
Food Retail                        0
On-farm Electricity Use            0
Food Packaging                     0
Agrifood Systems Waste Disposal    0
Food Processing                    0
Fertilizers Manufacturing          0
Manure left on Pasture             0
Fires in organic soils             0
Fires in humid tropical forests    0
Rural population                   0
Urban population                   0
Total Population - Male            0
Total Population - Female          0
total_emission                     0
Average Temperature °C             0
dtype: int64

In [18]:
duplicates=df_filtered.duplicated().sum()
duplicates

np.int64(0)

There are no duplicated data points 

In [19]:
# Count number of unique values in each column

df_filtered.nunique()

Area                                236
Year                                 31
Savanna fires                      3747
Forest fires                       2963
Rice Cultivation                   3635
Drained organic soils (CO2)        2147
Pesticides Manufacturing           1310
Food Transport                     6512
Forestland                          606
Net Forest conversion               443
Food Household Consumption         6359
Food Retail                        6459
On-farm Electricity Use            5322
Food Packaging                     3704
Agrifood Systems Waste Disposal    6419
Food Processing                    3542
Fertilizers Manufacturing          2973
Manure left on Pasture             5922
Fires in organic soils              126
Fires in humid tropical forests    1737
Rural population                   6611
Urban population                   6865
Total Population - Male            6884
Total Population - Female          6884
total_emission                     6899


There are only 236 different areas and 31 different years 